synapse/config/emailconfig.py

```diff
@@ -74,19 +74,33 @@ def read_config(self, config, **kwargs):
             "renew_at"
         )
 
-        email_trust_identity_server_for_password_resets = email_config.get(
-            "trust_identity_server_for_password_resets", False
+        self.email_threepid_behaviour = (
+            # Have Synapse handle the email sending if account_threepid_delegate
+            # is not defined
+            "remote"
+            if self.account_threepid_delegate
+            else "local"
         )
-        self.email_password_reset_behaviour = (
-            "remote" if email_trust_identity_server_for_password_resets else "local"
-        )
-        self.password_resets_were_disabled_due_to_email_config = False
-        if self.email_password_reset_behaviour == "local" and email_config == {}:
+        # Prior to Synapse v1.4.0, there used to be another option that defined whether Synapse
+        # would use an identity server to password reset tokens on its behalf. We now warn the
+        # user if they have this set and tell them to use the updated option.
+        # TODO: Eventually we want to remove the functionality of having an identity server
+        #  send tokens on behalf of the homeserver. At that point, we should remove this check
+        if config.get("trust_identity_server_for_password_resets", False) is True:
+            raise ConfigError(
+                'The config option "trust_identity_server_for_password_resets" '
+                'has been replaced by "account_threepid_delegate". Please '
+                "consult the default config at docs/sample_config.yaml for "
+                "details and update your config file."
+            )
+
+        self.local_threepid_emails_disabled_due_to_config = False
+        if self.email_threepid_behaviour == "local" and email_config == {}:
             # We cannot warn the user this has happened here
             # Instead do so when a user attempts to reset their password
-            self.password_resets_were_disabled_due_to_email_config = True
+            self.local_threepid_emails_disabled_due_to_config = True
```

Comment:
This doesn't seem to be used anywhere?

Reply:
Ah indeed, I forgot to replace a variable.